src/api/v4/native/leaflet-layer.js

```diff
@@ -68,7 +68,8 @@ var LeafletLayer = L.TileLayer.extend({
   },
 
   _onFeatureClick: function (internalEvent) {
-    triggerLayerFeatureEvent(Layer.events.FEATURE_CLICKED, internalEvent, this._layers);
+    var layer = this._layers.findById(internalEvent.layer.id);
+    triggerLayerFeatureEvent(Layer.events.FEATURE_CLICKED, internalEvent, layer);
```

Comment:
We're mixing in this PR stuff related to interactivity along with the layer order, right? Correct me if I'm wrong.

Reply:
If we're adding more stuff apart from layer ordering, please write that down in the issue.